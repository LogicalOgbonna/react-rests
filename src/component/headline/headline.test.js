import React from "react";
import { shallow } from 'enzyme';
import Headline from "./headline";
import { findByTestAttr, checkProps } from "../../utils/tests";

const setUp = (props = {}, state = {}) => {
    return shallow(<Headline {...props} />)
}

describe('<Headline /> Component', () => {

    describe('Checking Proptypes', () => {
        it("Should not throw a warning", () => {
            const expectedProps = {
                header: "Hello",
                description: "Test Description",
                user: {
                    name: "Arinze",
                    email: "arinzeogbo@gmail.com",
                    routes: [{
                        name: "Home",
                        url: "/user/me",
                        loggedIn: true
                    }, {
                        name: "Home",
                        url: "/user/me",
                        loggedIn: false
                    }]
                }
            }

            const propsError = checkProps(Headline, expectedProps);
            expect(propsError).toBeUndefined();
        })
    })

    describe('Have Props', () => {
        let component;
        beforeEach(() => {
            const props = {
                header: "Test Header",
                description: "Test Description"
            };
            component = setUp(props)
        });
        it("Should Render Without Crashing", () => {
            expect(findByTestAttr(component, "headerlineComponent").length).toBe(1)
        })
        it("Should Render <h1 />", () => {
            expect(findByTestAttr(component, "header").length).toBe(1)
        })
        it("Should Render <p />", () => {
            expect(findByTestAttr(component, "description").length).toBe(1)
        })
    });

    describe('Have No Props', () => {
        let component;
        beforeEach(() => {
            component = setUp();
        });
        it("Should Render Without Crashing", () => {
            expect(findByTestAttr(component, "headerComponent").length).toBe(0)
        });
        it("Should Not Render <h1 />", () => {
            expect(findByTestAttr(component, "header").length).toBe(0)
        })
        it("Should Not Render <p />", () => {
            expect(findByTestAttr(component, "description").length).toBe(0)
        })
    })
});
