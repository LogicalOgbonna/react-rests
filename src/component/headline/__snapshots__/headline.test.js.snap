// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Headline /> Component Have No Props should render <Headline /> Without Crashing 1`] = `undefined`;

exports[`<Headline /> Component Have Props should render <Headline /> Without Crashing 1`] = `undefined`;
